src/action/payment.js

```diff
@@ -122,6 +122,38 @@ class PaymentAction {
    */
   setAmount({ amount }) {
     this._store.payment.amount = amount;
+    this._store.payment.sendAll = false;
+  }
+
+  /**
+   * Set the payment amount to the max amount that can be sent. This
+   * is useful for people to move their coins off of the app.
+   * @return {Promise<undefined>}
+   */
+  async setMax() {
+    const timeout = setTimeout(() => {
+      return this._notification.display({
+        type: 'error',
+        msg: 'Setting max payment amount timed out!',
+      });
+    }, PAYMENT_TIMEOUT);
+    const { payment, balanceSatoshis, settings } = this._store;
+    let feeSat;
+    let amtSat = Math.floor(0.99 * balanceSatoshis);
+    while (!feeSat) {
+      try {
+        feeSat = await this._getFeeEstimateSat({
+          addr: payment.address,
+          amtSat,
+        });
+      } catch (err) {
+        amtSat = Math.floor(0.99 * amtSat);
+      } finally {
+        clearTimeout(timeout);
+      }
+    }
```

Comment:
Do we really want to poll `_getFeeEstimateSat` like this? Can't we just display a static max amount which has the fee deducted? I mean the on chain fee shouldn't change much depending on the amount right? And even if it does change slightly it shouldn't be a big issues as long as all of the funds are flushed to an external wallet.

Reply:
Yeah, I will rework this part.

The reason I wrote it this way is because the `estimateFee` gRPC call fails if given the total wallet balance, saying the amount is too high. But rather than polling, I'll just find a balance reduction that works for all reasonable wallet balances.

Reply:
What's the minimum sat amount hat can be sent in an on-chain tx? Couldn't we just query with that?

Reply:
In my testing I saw a fair amount of variation (once I saw $40 of fees) so I went with 98% of the total on-chain balance and that seems to work in the cases I tried.